### PR TITLE
Auxiliary space preconditioners fix

### PR DESCRIPTION
**Pull request by victorapm**
This PR fixes the data type of `cheby_fraction` used as input in the `SetChebySmoothingOptions` functions of AMS and ADS. It also fixes a comment about AMS in the documentation.

This PR addresses GitHub issues #857 and #858.

### CODE REVIEW

File: src/docs/usr-manual/solvers-ams.rst [hunk 1]
```diff
@@ -68,9 +68,9 @@ In addition to the above quantities, AMS can utilize the following (optional)
 information:
 
 * The Poisson matrices :math:`A_\alpha` and :math:`A_\beta`, corresponding to
```
Comment (rfalgout) [inline src/docs/usr-manual/solvers-ams.rst]: I suppose you can remove this change now from the PR

**Review by rfalgout — APPROVED** [review]
Looks good.  Should we add a few tests before merging this?

File: src/docs/usr-manual/solvers-ams.rst (hunk 1 above)
Comment (victorapm) [reply src/docs/usr-manual/solvers-ams.rst]: Yes, done at [79d5d84](https://github.com/hypre-space/hypre/pull/859/commits/79d5d84f3a91b65b9acb9888c15476d5b542b874)

**Comment by victorapm** [comment]
Thanks! I was thinking about another PR for the tests. It will involve adding a public `HYPRE` function, but I am fine adding the tests in this PR

**Comment by rfalgout** [comment]
> Thanks! I was thinking about another PR for the tests. It will involve adding a public `HYPRE` function, but I am fine adding the tests in this PR

It's fine to add tests in another PR.  Why do you need a new function (and what will it do)?

**Comment by victorapm** [comment]
Thank you! I need to add `HYPRE_AMSSetChebySmoothingOptions`. Note we already have this for ADS: https://github.com/hypre-space/hypre/blob/c3ce7b9f17cbf8019727dd4a4214bdd62815ef05/src/parcsr_ls/HYPRE_parcsr_ls.h#L2548

It is setting the input parameters to Chebyshev. Since we are already looking at this, I think it is a good idea to revise the interface. The function does not consider all input parameters that can affect Chebyshev setup, e.g., it does not have `cheby_variant`. In BoomerAMG, we have separate functions for setting each of the input parameters (order, variant, fraction...). Should we have the same for AMS and ADS? Or add a general function that sets all chebyshev parameters for BoomerAMG as well? I think we can consider these questions.

**Comment by victorapm** [comment]
Note: The function I was talking about already exists here https://github.com/hypre-space/hypre/blob/c3ce7b9f17cbf8019727dd4a4214bdd62815ef05/src/parcsr_ls/HYPRE_ams.c#L232

Just need to add it to `HYPRE_parcsr_ls.h`. Still, I think we can discuss the questions above :)

**Comment by rfalgout** [comment]
At a minimum, adding the existing function to the documentation makes sense.  We should bring some others into the discussion about changing the interface (and maybe add it to the agenda for our next meeting).

**Comment by victorapm** [comment]
Yes, makes sense. Thank you!

**Pull request merged**